src/os/bluestore/BlueStore.cc

```diff
@@ -739,7 +739,9 @@ void BlueStore::TwoQCache::trim(uint64_t onode_max, uint64_t buffer_max)
       Buffer *b = &*p;
       assert(b->is_clean());
       dout(20) << __func__ << " buffer_warm_in -> out " << *b << dendl;
+      assert(buffer_bytes >= b->length);
```

Comment:
@xiexingguo assert()s are cpu extensive , I don't think we need to do it within loop. Did you face this issue hitting ?
buffer_bytes is already > buffer_cache_size and it is unlikely that individual buffer length we are trimming > buffer_bytes.